pandas/src/parser/tokenizer.c

```diff
@@ -1072,169 +1038,177 @@ int tokenize_delim_customterm(parser_t *self, size_t line_limit)
                 self->state = SKIP_LINE;
-                if (c == self->lineterminator) {
+                if (IS_TERMINATOR(c)) {
                     END_LINE();
                 }
                 break;
-            }
-            else if (c == self->lineterminator) {
+            } else  if (IS_TERMINATOR(c)) {
+                if (self->skip_empty_lines) {
                 // \n\r possible?
-                if (self->skip_empty_lines)
-                {
                     self->file_lines++;
-                }
-                else
-                {
+                } else {
                     END_LINE();
                 }
                 break;
-            }
-            else if (c == self->commentchar) {
-                self->state = EAT_LINE_COMMENT;
+            } else if (IS_CARRIAGE(c)) {
+                if (self->skip_empty_lines) {
+                    self->file_lines++;
+                    self->state = EAT_CRNL_NOP;
+                } else {
+                    self->state = EAT_CRNL;
+                }
                 break;
-            }
-            else if (IS_WHITESPACE(c) && c != self->delimiter && self->skip_empty_lines)
-            {
-                self->state = WHITESPACE_LINE;
+            } else if (IS_WHITESPACE(c)) {
+                if (self->skip_empty_lines) {
+                    self->state = WHITESPACE_LINE;
+                } else {
+                    self->state = EAT_WHITESPACE;
+                }
                 break;
+            } else if (c == self->commentchar) {
+                self->state = EAT_LINE_COMMENT;
+                break;
+            } else {
+                // normal character - handle as START_FIELD
+                self->state = START_FIELD;
             }
-            /* normal character - handle as START_FIELD */
-            self->state = START_FIELD;
-            /* fallthru */
+            // fall through
         case START_FIELD:
-            /* expecting field */
-            if (c == self->lineterminator) {
+            // expecting field
+            if (IS_TERMINATOR(c)) {
                 END_FIELD();
                 END_LINE();
-                /* self->state = START_RECORD; */
-            }
-            else if (c == self->quotechar &&
-                     self->quoting != QUOTE_NONE) {
-                /* start quoted field */
+            } else if (IS_CARRIAGE(c)) {
+                END_FIELD();
+                self->state = EAT_CRNL;
+            } else if (c == self->quotechar &&
+                       self->quoting != QUOTE_NONE) {
+                // start quoted field
                 self->state = IN_QUOTED_FIELD;
-            }
-            else if (c == self->escapechar) {
-                /* possible escaped character */
+            } else if (c == self->escapechar) {
+                // possible escaped character
                 self->state = ESCAPED_CHAR;
-            }
-            else if (c == ' ' && self->skipinitialspace)
```

Comment:
Split the view on the diff, and you'll see it's still there.  This is one of those moments when Git's diff isn't super human friendly.